includes/wc-order-functions.php

```diff
@@ -909,6 +909,7 @@ function wc_cancel_unpaid_orders() {
 		}
 	}
 	wp_clear_scheduled_hook( 'woocommerce_cancel_unpaid_orders' );
+	$held_duration = apply_filters( 'woocommerce_cancel_unpaid_orders_duration', $held_duration );
```

Comment:
Rather wrap the filter around the line `$held_duration = get_option( 'woocommerce_hold_stock_minutes' );`

Reply:
Why do many create the new timing instead of using a recurring event? Details on https://developer.wordpress.org/plugins/cron/scheduling-wp-cron-events/